airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py

```diff
@@ -181,13 +184,25 @@ def get_client_secret(self) -> str:
     def get_refresh_token(self) -> str:
         return dpath.util.get(self._connector_config, self._refresh_token_config_path)
 
-    def set_refresh_token(self, new_refresh_token: str):
-        """Set the new refresh token value. The mutation of the connector_config object will emit an Airbyte control message.
+    def get_access_token_expiration_datetime(self) -> pendulum.DateTime:
+        return pendulum.parse(dpath.util.get(self._connector_config, self._access_token_expiration_datetime_config_path))
+
+
+    def _update_config_with_access_and_refresh_tokens(self, new_access_token: str, new_refresh_token: str, new_access_token_expiration_datetime: pendulum.DateTime):
+        """Update the connector configuration with new access and refresh token values. 
+        The mutation of the connector_config object will emit Airbyte control messages.
 
         Args:
+            new_access_token (str): The new access token value.
             new_refresh_token (str): The new refresh token value.
+            new_access_token_expiration_datetime (pendulum.DateTime): The new access token expiration date.
         """
+        # TODO alafanechere this will sequentially emit three control messages. 
+        # We should rework the observer/config mutation logic if we want to have atomic config updates in a single control message.
+        dpath.util.set(self._connector_config, self._access_token_config_path, new_access_token)
         dpath.util.set(self._connector_config, self._refresh_token_config_path, new_refresh_token)
+        dpath.util.set(self._connector_config, self._access_token_expiration_datetime_config_path, new_access_token_expiration_datetime)
```

Comment:
note: this means as currently implemented during the sync the platform will perform 3 separate config updates (api calls) for each of these pieces

Reply:
I discarded the use of `ObservedDict` and directly call a function from this method to emit the control message.